src/backend/distributed/executor/multi_utility.c

```diff
@@ -299,6 +300,39 @@ multi_ProcessUtility(Node *parsetree,
 								 " necessary users and roles.")));
 	}
 
+	/* due to an explain-hook limitation we have to special-case EXPLAIN EXECUTE */
+	if (IsA(parsetree, ExplainStmt) && IsA(((ExplainStmt*) parsetree)->query, Query))
+	{
+		ExplainStmt *explainStmt = (ExplainStmt *) parsetree;
+		Query *query = (Query *) explainStmt->query;
+
+		if (query->commandType == CMD_UTILITY &&
+			IsA(query->utilityStmt, ExecuteStmt))
+		{
+			ListCell *queryCell;
+			ExecuteStmt *execstmt = (ExecuteStmt *) query->utilityStmt;
+			PreparedStatement *entry = FetchPreparedStatement(execstmt->name, true);
+
+			/* copied from ExplainExecuteQuery, will never trigger if you used PREPARE */
+			if (!entry->plansource->fixed_result)
+			{
+				ereport(ERROR, (errmsg("EXPLAIN EXECUTE does not support variable-result"
+									   " cached plans")));
+			}
+
+			foreach(queryCell, entry->plansource->query_list)
+			{
+				Query *query = (Query *) lfirst(queryCell);
+				explainStmt->query = (Node *) query;
+
+				standard_ProcessUtility(parsetree, query_string, context,
```

Comment:
I don't understand query re-writing well enough to know what to do here. Should we do our special handling if any of the queries are distributed tables? Or only mess with the ones which hit distributed tables? Or maybe we only look at it if there's just one query? 

In what kinds of situations does query rewriting happen?